src/main/webapp/script.js

```diff
@@ -60,6 +60,27 @@ function passQuery() {
     });
 }
 
+function validateLocation() {
+  const datalist = document.getElementById('location');
+  const inputlist = document.getElementById('locationlist');
+
+  const options = datalist.options;
+  const typedSoFar = inputlist.value.toLowerCase();
+  let matchesAnOption = false;
+
+  for (const option of options) {
+    if (option.value.toLowerCase().includes(typedSoFar)) {
+      matchesAnOption = true;
+      inputlist.style.borderColor = '#767676'; // dark gray
+      inputlist.style.borderStyle = 'none none solid none';
+    }
```

Comment:
At this point you already have match so you can return from this function. there is no need to compare with other options. And if you do that then you don't even need matchesAnOption variable. When you fall out of the for loop you know have not matched any options.

Reply:
Very good point. Done